mypy/checker.py

```diff
@@ -1341,6 +1341,20 @@ def check_method_override_for_base_with_name(
                 self.msg.signature_incompatible_with_supertype(
                     defn.name(), name, base.name(), context)
 
+    def get_op_other_domain(self, tp: FunctionLike) -> Optional[Type]:
+        if isinstance(tp, CallableType):
+            if tp.arg_kinds and tp.arg_kinds[0] == ARG_POS:
```

Comment:
Do we care what happens if the user does `def __add__(self, *args: Foo) -> Foo` or something?

It's technically not illegal, and I think we even have one or two tests existing tests somewhere that check for this sort of thing in operator methods.

Reply:
I think we don't care for now, I would say this kind of situations is already rare, and detecting a rare unsafety due to extended domain is essentially `rare ** 2`.